service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/HearingBookingDetailsController.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import com.google.common.collect.ImmutableList;
+import io.swagger.annotations.Api;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.HearingBookingDetail;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.service.MapperService;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Objects;
+
+@Api
+@RestController
+@RequestMapping("/callback/add-hearing-booking")
+public class HearingBookingDetailsController {
+    private final MapperService mapperService;
+
+    @Autowired
+    public HearingBookingDetailsController(MapperService mapperService) {
+        this.mapperService = mapperService;
+    }
+
+    @PostMapping("/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody CallbackRequest callbackrequest) {
+        CaseDetails caseDetails = callbackrequest.getCaseDetails();
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetails.getData())
+            .errors(validate(caseDetails))
+            .build();
+    }
+
+    private List<String> validate(CaseDetails caseDetails) {
```

Comment:
Do we not do the same thing in respondent and applicant? Considering making this a util function and reusing.

Reply:
They are different for every one i think it is fine to keep it here and tied to the controller.

Reply:
Discussed and seems ok:)

Reply:
So as mentioned in the test Ideally it should be service/separate validator. Now this controller is doing a bit too much, which is visible from unit tests, but that's fine, it's simple validator.